pulp_ansible/app/serializers.py

```diff
@@ -132,7 +133,18 @@ def validate(self, data):
         data = super().validate(data)
 
         if data.get("requirements_file"):
-            parse_collections_requirements_file(data["requirements_file"])
+            parsed_file = parse_collections_requirements_file(data["requirements_file"])
+            urls_to_validate = [data["url"]]
+            for name, version, source in parsed_file:
+                if source:
+                    urls_to_validate.append(source)
+
+            for url in urls_to_validate:
+                result = re.findall(r"/v(\d)/collections", url)
```

Comment:
What do we do about the optional trailing slash or not? I think we either need to require it or not, but not allow both.

Reply:
Also what about cases where it's something like /something/api/v2/collection/more/stuff/v2/collectionmine/. This would also validate. So I think validating at the end is key.

Reply:
So maybe this check would instead be:

`if url.endswith('/v2/collections/`) or url.endswith('/v3/collections/`):`

or possibly `if url.endswith('/v2/collections`) or url.endswith('/v3/collections`):` depending on how the trailing slash is handled.